tests/testthat/test-metadata.R

```diff
@@ -8,34 +11,110 @@ extract_format <- function(.x) {
+        type = "text",
+        label = c("X Label", "Y Label"),
+        length = c(4, 4),
+        common = NA_character_,
+        format = NA_character_
+      )
+    )
+  )
   
-  extract_varlabel <- function(.x) {
+  extract_var_label <- function(.x) {
     vapply(.x, function(.x) attr(.x, "label"), character(1), USE.NAMES = FALSE) 
   }
   
-  df <- xportr_label(df, varmeta)
-  df_dput <- dput(df)
+  df_labeled_df <- df %>% xportr_label(df_meta)
+  metacoes_labeled_df <- df_labeled_df %>% xportr_label(metacore_meta, domain = "DOMAIN")
   
-  expect_equal(extract_varlabel(df), c("foo", "bar"))
-  expect_equal(df_dput,
-                structure(list(x = structure("a", label = "foo"),
-                               y = structure("b", label = "bar")),
-                          row.names = c(NA, -1L), class = "data.frame"))
+  expect_equal(extract_var_label(df_labeled_df), c("foo", "bar", ""))
+  expect_equal(extract_var_label(metacoes_labeled_df), c("X Label", "Y Label", ""))
+  expect_equal(
+    dput(df_labeled_df),
+    structure(
+      list(
+        x = structure("a", label = "foo"),
+        y = structure("b", label = "bar"),
+        variable = structure("value", label = "")
+      ),
+      row.names = c(NA, -1L),
+      `_xportr.df_arg_` = "df",
+      class = "data.frame"
+    )
+  )
+  expect_equal(
+    dput(metacoes_labeled_df),
+    structure(
+      list(
+        x = structure("a", label = "X Label"),
+        y = structure("b", label = "Y Label"),
+        variable = structure("value", label = "")
+      ),
+      row.names = c(NA, -1L),
+      `_xportr.df_arg_` = "DOMAIN",
+      class = "data.frame"
+    )
+  )
 })
 
 test_that("Dataset label", {
   df <- data.frame(x = "a", y = "b")
-  dfmeta <- data.frame(dataset  = "df", 
-                   label = "Label") 
+  renamed_dataset_df <- structure(
+    df,
+    `_xportr.df_arg_` = "CUSTOMNAME"
+  )
+
+  df_meta <- data.frame(dataset  = "df", label = "Label")
+  metacore_meta <- suppressWarnings(
+    metacore(
+      ds_spec = data.frame(
+        dataset  = c("df", "CUSTOMNAME", "DOMAINNAME"),
+        structure = "",
+        label = c("Label", "Custom Label", "Domain Label")
+      )
+    )
+  )
   
-  df <- xportr_df_label(df, dfmeta)    
-  expect_equal(attr(df, "label"), "Label") 
-  expect_equal(dput(df), structure(list(x = "a", y = "b"), class = "data.frame",
-                                   row.names = c(NA, -1L), label = "Label"))
+  df_spec_labeled_df <- xportr_df_label(df, df_meta)
+  piped_spec_labeled_df <- df %>% xportr_df_label(df_meta)
+  renamed_spec_labeled_df <- xportr_df_label(renamed_dataset_df, metacore_meta)
+  domain_spec_labeled_df <- xportr_df_label(renamed_dataset_df, metacore_meta, domain = "DOMAINNAME")
+
+  expect_equal(attr(df_spec_labeled_df, "label"), "Label") 
+  expect_equal(
+    dput(df_spec_labeled_df),
+    structure(list(x = "a", y = "b"), class = "data.frame", row.names = c(NA, -1L), label = "Label")
```

Comment:
could we add some more line breaks here and maybe even break these tests into smaller chunks if possible?

Reply:
Yeah, that makes sense. I will split them into smaller chunks. I also like your suggestion to have better test_that descriptions and the convention followed by the admiral is great and it's worth doing that kind of `function_name Test n: SINGLE TEST CASE` descriptions. I will create a task to do this across all our tests.

Reply:
Created issue #92 to make sure we standardize the test description across all of our tests. However, I was unable to add it to the board. Maybe you can add it there?